llvm/lib/CAS/OnDiskGraphDB.cpp

```diff
@@ -597,6 +597,14 @@ Error TempFile::keep(const Twine &Name) {
   Done = true;
   // Always try to close and rename.
   std::error_code RenameEC = sys::fs::rename(TmpName, Name);
+  if (RenameEC) {
```

Comment:
Nit: can we check for `errc::no_such_file_or_directory` (`ENOENT`) specifically? That should definitely work on POSIX and if it doesn't work on Windows I think we would discover that immediately because every cas test involving standalone files would fail.